webapp/channels/src/packages/mattermost-redux/src/actions/profiles_statuses_groups.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable no-console */
+
+import type {PostList, Post} from '@mattermost/types/posts';
+
+import {getProfilesByIds, getStatusesByIds} from 'mattermost-redux/actions/users';
+import {getCurrentChannelId, getCurrentUserId, getIsUserStatusesConfigEnabled} from 'mattermost-redux/selectors/entities/common';
+import type {ActionFunc} from 'mattermost-redux/types/actions';
+
+const pendingUserIdsForProfiles = new Set<string>();
+const MAX_USER_PROFILES_BATCH = 100;
+const USER_PROFILES_DURATION = 5 * 1000;
+let userProfilesIntervalId: NodeJS.Timeout | null = null;
+
+const pendingUserIdsForStatuses = new Set<string>();
+const MAX_USER_STATUSES_BUFFER = 200;
+const USER_STATUSES_DURATION = 5 * 1000;
+const USER_STATUSES_REQUEST_SURGE_THRESHOLD = 1000;
+let userStatusesIntervalId: NodeJS.Timeout | null = null;
+let haveUserStatusRequestsSurged = false;
+
+function addUserIdForProfiles(userId: string): ActionFunc<boolean> {
+    return (dispatch) => {
+        function getPendingProfilesById() {
+            console.log('addUserIdForProfiles start', pendingUserIdsForProfiles);
```

Comment:
Remember to remove this before this is merged. If you want to leave in the debug messages for testing when it hits Community, you could make it so that these are only printed when some variable is set on the Window